src/Microsoft.Azure.WebJobs/ParameterBindingData.cs

```diff
@@ -0,0 +1,87 @@
+{
+    /// <summary>
+    /// A reference type for supporting SDK-type bindings in out-of-process Function workers
+    /// </summary>
+    public class ParameterBindingData
+    {
+        /// <summary> Initializes a new instance of the <see cref="ParameterBindingData"/> class with the Content-Type set to application/json</summary>
+        /// <param name="source"> Identifies the extension this event is coming from </param>
+        /// <param name="jsonSerializableData"> An object containing any required information to hydrate an SDK-type object in the out-of-process worker </param>
+        /// <param name="dataSerializationType"> The type to use when serializing the data.
+        /// If not specified, <see cref="object.GetType()"/> will be used on <paramref name="jsonSerializableData"/>.</param>
+        /// <exception cref="ArgumentNullException">Throws when source is null.</exception>
+        /// <exception cref="ArgumentException">Throws when jsonSerializableData is of type BinaryData.</exception>
+        public ParameterBindingData(string source, object jsonSerializableData, Type dataSerializationType = default)
+        {
+            if (jsonSerializableData is BinaryData)
+            {
+                throw new ArgumentException("This constructor does not support BinaryData. Use the constructor that takes a BinaryData instance.");
+            }
+
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            Version = "1.0";
+            Source = source;
+            ContentType = "application/json";
+            Content = new BinaryData(jsonSerializableData, type: dataSerializationType ?? jsonSerializableData?.GetType());
+        }
+
+        /// <summary> Initializes a new instance of the <see cref="ParameterBindingData"/> class using binary event data.</summary>
+        /// <param name="source"> Identifies the extension this event is coming from </param>
+        /// <param name="content"> Binary data containing any required information to hydrate an SDK-type object in the out-of-process worker </param>
+        /// <param name="contentType"> Content type of the payload. A content type different from "application/json" should be specified if payload is not JSON. </param>
+        /// <exception cref="ArgumentNullException">Throws if source, content or contentType is null.</exception>
+        public ParameterBindingData(string source, BinaryData content, string contentType)
+        {
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            if (contentType is null)
+            {
+                throw new ArgumentNullException(nameof(contentType));
+            }
+
+            if (content is null)
+            {
+                throw new ArgumentNullException(nameof(content));
+            }
+
+            Version = "1.0";
```

Comment:
Looks like this is hardcoded to 1, and the actual producer/consumer of this type can't control the version? Is this the version of this type, or the version of the contents? How is this version intended to be used?

Reply:
The version is the schema version, so this type. Ideally consumers are checking for this for backwards compatibility. So whenever we make breaking changes to this type, we can bump the version. But I'm open to better suggestions for versioning!

Reply:
Had a sync with Fabio about this, and agree that it should be the version of the content and not the schema. Updating this so that version is provided by the provider